lang/java/protobuf/src/test/java/org/apache/avro/protobuf/TestProtobuf.java

```diff
@@ -87,6 +87,26 @@ public void testMessage() throws Exception {
     assertEquals(foo, o);
   }
 
+  @Test
+  public void testMessageWithEmptyArray() throws Exception {
+    Foo foo = Foo.newBuilder().setInt32(5).setBool(true).build();
+    ByteArrayOutputStream bao = new ByteArrayOutputStream();
+    ProtobufDatumWriter<Foo> w = new ProtobufDatumWriter<>(Foo.class);
+    Encoder e = EncoderFactory.get().binaryEncoder(bao, null);
+    w.write(foo, e);
+    e.flush();
+    Object o = new ProtobufDatumReader<>(Foo.class).read(null,
+        DecoderFactory.get().binaryDecoder(new ByteArrayInputStream(bao.toByteArray()), null));
+
+    assertEquals(foo, o);
```

Comment:
This test fails, since 'foo' has many unset fields, while in 'o' they've been set to their default value.  Avro's protobuf support doesn't preserve the set/unset status of fields.

Reply:
I see.
Then I don't think it makes sense to check for the full equality of "foo" and "o", and instead, I should just check the set fields are equal and "o"'s array fields are empty.